backend/src/test/java/de/neuefische/driver/backend/services/DeliveryServiceTest.java

```diff
@@ -45,4 +54,46 @@ void getDeliveries_shouldReturn_listOfDeliveries() {
         List<Delivery> expected = new ArrayList<>(List.of(test));
         assertEquals(expected, actual);
     }
+
+
+    @Test
+    void getDeliveryById_shouldReturnRequestedDelivery() {
+        Delivery requested = createTestDeliveryInstance();
+
+        Mockito.when(deliveryRepo.getDeliveryById(testIdOne))
+                .thenReturn(requested);
+
+        Delivery actual = deliveryService.getDeliveryById(testIdOne);
+
+        verify(deliveryRepo).getDeliveryById(testIdOne);
+        assertEquals(createTestDeliveryInstance(), actual);
+    }
+
+    @Test
+    void getDeliveryById_shouldThrowException_whenInvalidId(){
+        Mockito.when(deliveryRepo.getDeliveryById(testIdOne))
+                .thenReturn(null);
+        try {
+            deliveryService.getDeliveryById(testIdOne);
+            fail();
+        }catch(NoSuchElementException e){
+            assertTrue(true);
+        }
+        verify(deliveryRepo).getDeliveryById(testIdOne);
+    }
```

Comment:
Hat das einen expliziten Grund, warum ihr in diesem Test im Gegensatz zum dazugehörigen Integrationstest die Logik im catch() clause umgedreht habt?